draft-ietf-anima-brski-prm.md

```diff
@@ -1694,117 +1751,153 @@ This results in a signed CA certificate(s) object (JSON-in-JWS), the CA certific
+
+{{exchangesfig_uc2_6}} shows the provisioning of the voucher to the pledge and the following subsections describe the corresponding artifacts.
 
 ~~~~ aasvg
 +--------+    +------------+    +-----------+    +--------+    +------+
-| Pledge |    | Registrar- |    |  Domain   |    | Domain |    | MASA |
-|        |    |   Agent    |    | Registrar |    |   CA   |    |      |
+| Pledge |    | Registrar- |    |  Domain   |    |  Key   |    | MASA |
+|        |    |   Agent    |    | Registrar |    | Infra. |    |      |
 +--------+    +------------+    +-----------+    +--------+    +------+
  |                  |                 |                 |   Internet |
  ~                  ~                 ~                 ~            ~
 (6) Supply Voucher to Pledge
  ~                  ~                 ~                 ~            ~
  |                  |                 |                 |            |
  |<----opt. TLS---->|                 |                 |            |
- |<-----Voucher-----|                 |                 |            |
+ |<-----Voucher''---|                 |                 |            |
  |------vStatus---->|                 |                 |            |
  |                  |                 |                 |            |
  ~                  ~                 ~                 ~            ~
 ~~~~
 {: #exchangesfig_uc2_6 title="Voucher exchange" artwork-align="center"}
 
-### Request Artifact: Voucher
+The Registrar-Agent SHALL supply the voucher to the pledge via HTTP(S) POST to the pledge endpoint at `/.well-known/brski/svr`.
+The request body MUST contain the Registrar-Countersigned Voucher (Voucher') artifact previously acquired from the domain registrar as defined in {{rcv-artifact}}.
+In the request header, the Content-Type field MUST be set to `application/voucher-jws+json` as defined in [I-D.ietf-anima-jws-voucher] and the Accept field SHOULD be set to `application/jose+json`.
+
+Upon receiving the voucher, the pledge SHALL perform the signature verification in the following order:
+
+1. Verify the MASA signature as described in {{Section 5.6.1 of !RFC8995}} against the pre-installed manufacturer trust anchor (e.g., IDevID).
+2. Provisionally install the trust anchor contained in the `pinned-domain-cert` field of the voucher.
+3. Validate the registrar LDevID certificate received in the `agent-provided-proximity-registrar-cert` field of the previously received tPVR artifact using the pinned domain certificate.
+4. Verify registrar signature of the Voucher' artifact similar as described in {{Section 5.6.1 of !RFC8995}}, but use the pinned domain certificate instead of the MASA certificate for the verification.
+
+BRSKI-PRM instroduces Steps 3 and 4 to verify the provisionally accepted registrar LDevID certificate from the tPVR parameter `agent-provided-proximity-registrar-cert` (see {{agt_prx}}). TODO(what about agent-signed data verification?)
```

Comment:
In 7.1.1 it says

> can only accept the parameters provisionally until it receives the voucher

One of these parameters in the agent-signed data. Is it actually relevant for the pledge? It feels like it is actually only relevant for registrar and MASA to assert agent-proximity. The pledge does not verify or accept, just copy to the PVR.

Reply:
Yes, it cannot validate it. It is intended to inform the registrar which registrar-agent has been used to onboard the pledge.

Reply:
I clarified that the provisionally accept is about the registrar LDevID cert

> for the object security within the authenticated self-contained objects that in BRSKI-PRM replace the direct TLS connection to the registrar in BRSKI {{!RFC8995}}

Reply:
Yes, agent-signed data is just forwarded by the pledge in the PVR